### PR TITLE
Change position of Publish, Delete and Preview buttons

### PR DESCRIPTION
**Pull request by DilwoarH**
## Changes
- Changes the position of the publish, delete and preview buttons

The position of the "Delete" Button on the Step by step overview page makes it feel like the default button to click.

We should reorder the buttons so that publishers don't keep clicking on Delete by accident.

## Preview
jul-30-2018 16-36-23

Ticket: https://trello.com/c/3HrLsnG4/728-change-position-of-publish-delete-and-preview-buttons

### CODE REVIEW

File: app/views/navigation_rules/edit.html.erb [hunk 1]
```diff
@@ -25,6 +25,8 @@
   action: 'Choose on-page side navigation'
 %>
 
+<%= render 'step_by_step_pages/nav', step_by_step_page: @step_by_step_page, active:"choose-navigation" %>
```
Comment (leenagupte) [inline app/views/navigation_rules/edit.html.erb]: Nitpick that @andysellick originally spotted: There should be a space between `:` and `"`

File: app/views/shared/steps/_form_notice.html.erb [hunk 2]
```diff
@@ -4,7 +4,7 @@
 %>
 <% if message %>
   <div class="row">
-    <div class="col-md-7">
+    <div class="col-md-12">
```
Comment (leenagupte) [inline app/views/shared/steps/_form_notice.html.erb]: How is this change related to the change to tabbed navigation?

File: app/views/step_by_step_pages/_form.html.erb [hunk 3]
```diff
@@ -31,7 +31,7 @@
   </div>
 
   <div class="form-group">
-    <%= form.submit "Save and continue", class: "btn btn-primary" %>
+    <%= form.submit "Save", class: "btn btn-primary" %>
```
Comment (leenagupte) [inline app/views/step_by_step_pages/_form.html.erb]: This should be moved into a separate commit as renaming the button is related to the change of navigation styles.

File: app/views/step_by_step_pages/edit.html.erb [hunk 4]
```diff
@@ -9,7 +9,7 @@
       href: @step_by_step_page
     },
     {
-      text: 'Edit nav'
+      text: 'Edit title, intro or slug'
```
Comment (leenagupte) [inline app/views/step_by_step_pages/edit.html.erb]: Is this change related to the navigation changes?

File: app/views/step_by_step_pages/edit.html.erb [hunk 5]
```diff
@@ -20,7 +20,9 @@
 
 <%= render 'shared/steps/heading',
   step_nav: @step_by_step_page.title,
-  action: 'Edit step by step navigation'
+  action: 'Edit title, intro or slug'
```
Comment (leenagupte) [inline app/views/step_by_step_pages/edit.html.erb]: Is this change related to the navigation changes?

File: app/views/step_by_step_pages/show.html.erb [hunk 6]
```diff
@@ -17,20 +17,15 @@
 
 <%= render 'shared/steps/form_notice', message: notice, status: "success" %>
 
-<h1><%= @step_by_step_page.title %></h1>
+<%= render 'shared/steps/heading',
+  step_nav: @step_by_step_page.title,
```
Comment (leenagupte) [inline app/views/step_by_step_pages/show.html.erb]: Is this just changing how the title is rendered?

File: app/views/step_by_step_pages/show.html.erb [hunk 7]
```diff
@@ -17,20 +17,15 @@
...
-  <li><%= link_to 'Choose navigation', step_by_step_page_navigation_rules_path(@step_by_step_page), class: "btn btn-primary" %></li>
-  <% if @step_by_step_page.steps.length > 0 %>
-    <li><%= link_to "Reorder steps", step_by_step_page_reorder_path(@step_by_step_page), class: "btn btn-primary" %></li>
-  <% end %>
-</ul>
+<%= render 'nav', step_by_step_page: @step_by_step_page, active:"edit-steps" %>
 
 <div class="row">
-  <div class="col-md-7">
+  <div class="col-md-12">
```
Comment (leenagupte) [inline app/views/step_by_step_pages/show.html.erb]: How is this change related to the change to tabbed navigation?

File: spec/features/managing_step_by_step_pages_spec.rb [hunk 8]
```diff
@@ -145,7 +145,7 @@ def and_I_fill_in_the_form
     fill_in "Introduction", with: "Learn how you can bake a cake"
     fill_in "Meta description", with: "How to bake a cake - learn how you can bake a cake"
 
-    click_on "Save and continue"
+    click_on "Save"
```
Comment (leenagupte) [inline spec/features/managing_step_by_step_pages_spec.rb]: These test updates should be moved into a separate commit as renaming the button isn't related to the change of navigation styles.

File: app/views/step_by_step_pages/show.html.erb [hunk 9]
```diff
@@ -60,18 +60,13 @@
...
-      <% else %>
-        <li><%= link_to 'Delete', @step_by_step_page, method: :delete, data: { confirm: 'Delete this step by step page?' }, class: "btn btn-danger" %></li>
-      <% end %>
-      <% if @step_by_step_page.has_draft? %>
-        <li><%= link_to 'Publish', step_by_step_page_publish_path(@step_by_step_page), class: "btn btn-primary" %></li>
-      <% end %>
-    </ul>
+    <div class="col-md-2 col-md-offset-10">
+      <ul class="list-unstyled step-by-step-actions text-right">
+        <li><%= link_to 'Add a new step', new_step_by_step_page_step_path(@step_by_step_page), class: "btn btn-primary btn-block" %></li>
```
Comment (leenagupte) [inline app/views/step_by_step_pages/show.html.erb]: Are these buttons being added, or just being moved on the page?

Either way, I think this commit should just focus on the movement of the "Publish", "Unpublish" and "Delete" buttons

File: spec/features/managing_step_by_step_pages_spec.rb [hunk 10]
```diff
@@ -47,25 +47,28 @@
 
   scenario "User deletes a step by step page", js: true do
     given_there_is_a_step_by_step_page_with_steps
-    and_I_visit_the_index_page
```
Comment (leenagupte) [inline spec/features/managing_step_by_step_pages_spec.rb]: Don't you need this step to visit the step-by-step overview page?

Should the test read:
```
and_I_visit_the_index_page
and_I_visit_the_publish_or_delete_page
```

So that it mirrors the action that the user takes?

File: spec/features/managing_step_by_step_pages_spec.rb [hunk 11]
```diff
@@ -19,7 +19,7 @@
   scenario "User creates a new step by step page" do
     when_I_visit_the_new_step_by_step_form
     and_I_fill_in_the_form
-    then_I_see_delete_and_publish_buttons
+    and_I_see_a_page_created_success_notice
```
Comment (leenagupte) [inline spec/features/managing_step_by_step_pages_spec.rb]: 💯

File: spec/features/managing_step_by_step_pages_spec.rb [hunk 12]
```diff
@@ -47,25 +47,28 @@
...
-    and_I_visit_the_index_page
+    and_I_visit_the_publish_or_delete_page
     and_I_delete_the_step_by_step_page
     then_the_draft_is_discarded
     and_the_page_is_deleted
   end
 
   scenario "User publishes a page" do
     given_there_is_a_step_by_step_page_with_steps
+    and_I_visit_the_publish_or_delete_page
```
Comment (leenagupte) [inline spec/features/managing_step_by_step_pages_spec.rb]: Should this test have `when_I_view_the_step_by_step_page` as the step before?

File: app/views/step_by_step_pages/publish_or_delete.html.erb [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+%>
+
+<%= render 'nav', step_by_step_page: @step_by_step_page, active:"publish-or-delete" %>
+
+<div class="row publish-or-delete">
+  <div class="col-md-12">
+    <% if @step_by_step_page.has_draft? %>
+        <div class="panel panel-default">
+            <div class="panel-body">
+                <%= link_to 'Publish changes', step_by_step_page_publish_path(@step_by_step_page), class: "btn btn-primary publish-button" %>
```
Comment (leenagupte) [inline app/views/step_by_step_pages/publish_or_delete.html.erb]: I'd consider separating out moving the buttons from renaming them.

File: app/assets/stylesheets/steps.scss [hunk 14]
```diff
@@ -20,6 +24,16 @@ pre.markdown-example {
   white-space: pre-wrap;
 }
 
+.nav.step-by-step-tabs {
+  margin-bottom: 2em;
+
+  @media screen and (min-width: 769px) and (max-width: 1000px) {
+    a {
```
Comment (andysellick) [inline app/assets/stylesheets/steps.scss]: I think it would be useful if you got into the habit of only styling with classes, not using element selectors like this (as this is the GOV.UK way). I know it's more effort, because you have to apply a class to all the extra elements, but it's worth it in the long term for maintainability.

File: app/views/step_by_step_pages/_nav.html.erb [hunk 15]
```diff
@@ -0,0 +1,35 @@
+<% 
+  unless defined? active
+    active = ''
+  end
+%>
+
+<ul class="nav nav-tabs nav-justified step-by-step-tabs">
+  <li 
+    class="<%= "active" if active == 'edit-steps' %>"
+  >
```
Comment (andysellick) [inline app/views/step_by_step_pages/_nav.html.erb]: As there's only one attribute on these LIs I'd put them all on one line.

File: app/views/step_by_step_pages/_nav.html.erb [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+    <%= link_to 'Edit steps', @step_by_step_page %>
+  </li>
+  <% if @step_by_step_page.steps.length > 0 %>
+    <li
+      class="<%= "active" if active == 'reorder-steps' %>"
+    >
+        <%= link_to "Reorder steps", step_by_step_page_reorder_path(@step_by_step_page)%>
```
Comment (andysellick) [inline app/views/step_by_step_pages/_nav.html.erb]: Nitpick: indentation.

File: app/views/step_by_step_pages/publish_or_delete.html.erb [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+  step_nav: @step_by_step_page.title,
+  action: 'Publish or delete'
+%>
+
+<%= render 'nav', step_by_step_page: @step_by_step_page, active:"publish-or-delete" %>
+
+<div class="row publish-or-delete">
+  <div class="col-md-12">
+    <% if @step_by_step_page.has_draft? %>
+        <div class="panel panel-default">
```
Comment (andysellick) [inline app/views/step_by_step_pages/publish_or_delete.html.erb]: Again I think the indentation here is inconsistent - looks like it starts a few lines above with two spaces and suddenly switches to 4. Give me a shout when we're both in the office and we'll check your editor's settings.

File: app/views/step_by_step_pages/publish_or_delete.html.erb [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+            <div class="panel-body">
+                <%= link_to 'Publish changes', step_by_step_page_publish_path(@step_by_step_page), class: "btn btn-primary publish-button" %>
+                <p>You will need to tag this step by step to a mainstream browse category and a taxonomy topic after publishing</p>
+            </div>
+        </div>
+    <% end %>
+
```
Comment (andysellick) [inline app/views/step_by_step_pages/publish_or_delete.html.erb]: Nitpick: double line space.

File: app/views/step_by_step_pages/edit.html.erb (hunk 4 above)
Comment (DilwoarH) [reply app/views/step_by_step_pages/edit.html.erb]: hey, yes as part of the tabbing change - the rename was required.

File: app/views/step_by_step_pages/show.html.erb (hunk 6 above)
Comment (DilwoarH) [reply app/views/step_by_step_pages/show.html.erb]: This makes the heading consistent with the other tabs, before it was not noticiable as you were transistioning across multiple pages.

File: app/views/step_by_step_pages/show.html.erb (hunk 7 above)
Comment (DilwoarH) [reply app/views/step_by_step_pages/show.html.erb]: The tabbed nav is full width, the content, therefore, looks weird and out of place when it is not full width

File: spec/features/managing_step_by_step_pages_spec.rb (hunk 8 above)
Comment (DilwoarH) [reply spec/features/managing_step_by_step_pages_spec.rb]: This needed to be renamed as every other tab has "Save", this was the only one that was "Save and continue"

File: app/views/shared/steps/_form_notice.html.erb (hunk 2 above)
Comment (DilwoarH) [reply app/views/shared/steps/_form_notice.html.erb]: the tabs now span over 12 columns, therefore, the error message should follow a consistent width

File: app/views/step_by_step_pages/edit.html.erb (hunk 4 above)
Comment (leenagupte) [reply app/views/step_by_step_pages/edit.html.erb]: Why?

File: app/views/step_by_step_pages/show.html.erb (hunk 6 above)
Comment (leenagupte) [reply app/views/step_by_step_pages/show.html.erb]: So perhaps do this first, separately?

File: app/views/step_by_step_pages/show.html.erb (hunk 7 above)
Comment (leenagupte) [reply app/views/step_by_step_pages/show.html.erb]: Can you add this explanation to the commit message?

File: spec/features/managing_step_by_step_pages_spec.rb (hunk 8 above)
Comment (leenagupte) [reply spec/features/managing_step_by_step_pages_spec.rb]: Making the contents of the tabs consistent isn't the same as changing the style of the navigation so it could be separated out.

File: app/views/step_by_step_pages/publish_or_delete.html.erb [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+    },
+    {
+      text: 'Publish or delete'
+    }
+  ]
+%>
+
+<%= render 'shared/steps/step_breadcrumb', links: links %>
+
+<%= render 'shared/steps/form_notice', message: notice, status: "success" %>
```
Comment (leenagupte) [inline app/views/step_by_step_pages/publish_or_delete.html.erb]: Where/when is `notice` being set?

File: app/views/step_by_step_pages/publish_or_delete.html.erb [hunk 20]
```diff
@@ -0,0 +1,50 @@
+<%
+  links = [
```
Comment (leenagupte) [inline app/views/step_by_step_pages/publish_or_delete.html.erb]: This is not related to the changes here, but I've noticed that the links for all of the step-by-step screens are being defined in the view. This is something that should probably be moved into a presenter sometime.

File: app/views/shared/steps/_form_notice.html.erb (hunk 2 above)
Comment (DilwoarH) [reply app/views/shared/steps/_form_notice.html.erb]: https://github.com/alphagov/collections-publisher/pull/438/commits/28c55b952e4ba32934bc99f88360378efa4508a1

File: app/views/step_by_step_pages/_form.html.erb (hunk 3 above)
Comment (DilwoarH) [reply app/views/step_by_step_pages/_form.html.erb]: Ref: https://github.com/alphagov/collections-publisher/pull/438/commits/cb3c7538bc7512aa3b3ec69bcc65f5babffbce07

File: spec/features/managing_step_by_step_pages_spec.rb (hunk 12 above)
Comment (DilwoarH) [reply spec/features/managing_step_by_step_pages_spec.rb]: Done
